src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2168,6 +2181,107 @@ private CallResult ExecuteCall<TTracingInstructions>(EvmState vmState, ReadOnlyM
         return GetFailureReturn<TTracingInstructions>(gasAvailable, exceptionType);
     }
 
+    [SkipLocalsInit]
+    private EvmExceptionType InstructionAuth<TTracingInstructions>(EvmState vmState, ref EvmStack<TTracingInstructions> stack, ref long gasAvailable, IReleaseSpec spec) where TTracingInstructions : struct, IIsTracing
+    {
+        if (!spec.AuthCallsEnabled) return EvmExceptionType.BadInstruction;
+        Address authority = stack.PopAddress();
+
+        if (authority is null) return EvmExceptionType.StackUnderflow;
+
+        if (!stack.PopUInt256(out UInt256 offset)) return EvmExceptionType.StackUnderflow;
+        if (!stack.PopUInt256(out UInt256 length)) return EvmExceptionType.StackUnderflow;
```

Comment:
We have 

```
        SkipInit(out UInt256 a);
        SkipInit(out UInt256 b);
        SkipInit(out UInt256 c);
        SkipInit(out UInt256 result);
        SkipInit(out StorageCell storageCell);
```

Can those be reused rather that allocating more on stack? Passing them by ref here? This can be also used for other methods in VM?

@benaadams

Reply:
Not really as it will allocate more stack to pass the parameters to the method; abi calling convetion means only have about 2 registers for an instance generic (first register is `this` pointer, second is generic type)

<img width="668" alt="image" src="https://github.com/NethermindEth/nethermind/assets/1142958/2ba51c0a-5829-4ed6-be4f-a4a37aac928c">

Reply:
I assume AUTH and AUTHCALL won't be so heavily used that the stack amount should be significant on them

Reply:
One UInt256 is 32 bytes, but passing it by ref should be only 8 bytes, no?